mailpoet/lib/Automation/Engine/Mappers/WorkflowMapper.php

```diff
@@ -0,0 +1,47 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Engine\Mappers;
+
+use DateTimeImmutable;
+use MailPoet\Automation\Engine\Data\NextStep;
+use MailPoet\Automation\Engine\Data\Step;
+use MailPoet\Automation\Engine\Data\Workflow;
+use MailPoet\Automation\Engine\Storage\WorkflowStatisticsStorage;
+
+class WorkflowMapper {
```

Comment:
Maybe we should extract `WorkflowsGetEndpoint::buildWorkflow()` as well.

Under REST point of view, I would think `steps` should be also in `WorkflowsGetEndpoint`, but not so important right now.

Maybe a `public function buildReducedWorkflow(Workflow $workflow, WorkflowStatistics $statistics = null)`.

My current concern is, at some point someone will just replace the private builder there with this one and we run a lot of db queries for the statistics instead of one.

Reply:
Yes! Implemented in: https://github.com/mailpoet/mailpoet/pull/4376/commits/03f9750aebf0e8b88ed8c3ff8303abfea88a7380

> Under REST point of view, I would think steps should be also in WorkflowsGetEndpoint, but not so important right now.

This is actually fine. We've been dealing with these issues in the past, and what seems to allow this difference is that these are different REST resources, in fact:

```bash
/workflows
/workflows/<id>
```

The former one always returns a list (can have filters, paging, etc.), and that can be a reduced item, the latter one always returns a full single `Workflow` resource, no matter the method (GET, POST, PUT, ...). Thinking about this, we seem to return `null` for `DELETE`. We could as well return the just deleted workflow, I guess...

Reply:
Can we return a `204 No Content` for the `DELETE` endpoint? That would solve the question what status the returned workflow should have.

Reply:
@websupporter Our API will always return something like `{"data": null}`, I think. It kind of is a "content," but I guess combining that with 204 would do no harm...? Returning the just-deleted entity is also an option, are there any use cases where that could be useful? E.g. for the UI to render something after the deletion? 🤔

(In any case I'd add it to the checklist rather than appending it here.)

Reply:
> E.g. for the UI to render something after the deletion

Yes, maybe something like a notice `Workflow "<name>" has been deleted`

> In any case I'd add it to the checklist rather than appending it here
:+1:

Reply:
:+1: I agree, I think usually, you'll probably have the name just before deleting, but yeah, this could facilitate some UIs. Added to the checklist.